### PR TITLE
Remove 'bazel/$package:$tag' naming format requirements from add_apt_key, install_pkgs, and download_pkgs rules

### PR DESCRIPTION
**Pull request by ArthurRab**

### CODE REVIEW

File: package_managers/download_pkgs.bzl [hunk 1]
```diff
@@ -114,6 +113,12 @@ _attrs = {
         mandatory = True,
     ),
     "additional_repos": attr.string_list(),
+
```
Comment (xingao267) [inline package_managers/download_pkgs.bzl]: extra line

File: package_managers/install_pkgs.bzl [hunk 2]
```diff
@@ -176,6 +180,7 @@ install = struct(
     implementation = _impl,
 )
 
+
```
Comment (xingao267) [inline package_managers/install_pkgs.bzl]: extra line

File: util/image_loader.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+temp=$(docker load --input $1)
+if [ `echo $temp | cut -d " " -f3` = "ID:" ]
+then
+  image_name=$(echo $temp | cut -d ":" -f3)
+else
+  image_name=$(echo $temp | cut -d " " -f3)
+fi
+echo $image_name
```
Comment (xingao267) [inline util/image_loader.sh]: Add new line at the end of file

File: util/image_loader.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+temp=$(docker load --input $1)
```
Comment (xingao267) [inline util/image_loader.sh]: Add comment which is image tag, which is id

File: util/run.bzl [hunk 5]
```diff
@@ -77,10 +73,6 @@ _extract_attrs = {
         single_file = True,
         cfg = "target",
     ),
-    "image_name": attr.string(
```
Comment (xingao267) [inline util/run.bzl]: line 134 below should be removed. No need to do "image_tar, intermediate_image = _rename_image(image, name)" in container_run_and_extract() anymore.

**Comment by xingao267** [comment]
This PR continues review from #2. In #2, we pursued a strategy to untar the image tarball and extract the ID from there. The concern with that is the tarball could be very large, and the untar can be slow. In #3, we get image ID/tag from the "docker load " command, which is run anyway.

File: util/run.bzl [hunk 6]
```diff
@@ -189,18 +155,26 @@ def _commit_impl(ctx):
...
 
-_run_and_commit = rule(
+"""Runs commands in a container and commits the container to a new image.
+
+This rule runs a set of commands in a given image, waits for the commands
+to finish, and then commits the container to a new image.
+
+
+Args:
+    image_name: Name of the image to run commands on.
```
[the remark this answers is not in the document]
Comment (ArthurRab) [reply util/run.bzl]: Done

File: util/run.bzl [hunk 7]
```diff
@@ -189,18 +155,26 @@ def _commit_impl(ctx):
...
+    image_tar: Tarball of image to run commands on.
```
[the remark this answers is not in the document]
Comment (ArthurRab) [reply util/run.bzl]: Done

File: util/run.bzl [hunk 8]
```diff
@@ -189,18 +155,26 @@ def _commit_impl(ctx):
...
+    commands: A list of commands to run (sequentially) in the container.
+    _run_tpl: Template for generated script to run docker commands.
+"""
```
[the remark this answers is not in the document]
Comment (ArthurRab) [reply util/run.bzl]: Done

File: util/run.bzl (hunk 6 above)
[the remark this answers is not in the document]
Comment (xingao267) [reply util/run.bzl]: Maybe reverted by accident?

File: util/run.bzl (hunk 7 above)
[the remark this answers is not in the document]
Comment (xingao267) [reply util/run.bzl]: ditto

File: util/run.bzl (hunk 8 above)
[the remark this answers is not in the document]
Comment (xingao267) [reply util/run.bzl]: ditto

File: util/run.bzl (hunk 6 above)
[the remark this answers is not in the document]
Comment (ArthurRab) [reply util/run.bzl]: woops fixed

File: util/run.bzl (hunk 7 above)
[the remark this answers is not in the document]
Comment (ArthurRab) [reply util/run.bzl]: fixed

File: util/run.bzl (hunk 8 above)
[the remark this answers is not in the document]
Comment (ArthurRab) [reply util/run.bzl]: fixed

**Pull request merged**